geemap/toolbar.py

```diff
@@ -379,31 +395,33 @@ def toolbar_btn_click(change):
 
     toolbar_button.observe(toolbar_btn_click, "value")
 
-    def close_btn_click(change):
-        if change["new"]:
-            toolbar_button.value = False
-            if m is not None:
-                if hasattr(m, "inspector_mode"):
-                    delattr(m, "inspector_mode")
-                m.toolbar_reset()
-                if m.tool_control is not None and m.tool_control in m.controls:
-                    m.remove_control(m.tool_control)
-                    m.tool_control = None
-                m.default_style = {"cursor": "default"}
+    def cleanup():
+        toolbar_button.value = False
+        if m is not None:
+            if hasattr(m, "inspector_mode"):
+                delattr(m, "inspector_mode")
+            if m.tool_control is not None and m.tool_control in m.controls:
+                m.remove_control(m.tool_control)
+                m.tool_control = None
+            m.default_style = {"cursor": "default"}
 
-                m.marker_cluster.markers = []
-                m.pixel_values = []
-                marker_cluster_layer = m.find_layer("Inspector Markers")
-                if marker_cluster_layer is not None:
-                    m.remove_layer(marker_cluster_layer)
+            m.marker_cluster.markers = []
+            m.pixel_values = []
+            marker_cluster_layer = m.find_layer("Inspector Markers")
+            if marker_cluster_layer is not None:
+                m.remove_layer(marker_cluster_layer)
+
+            if hasattr(m, "pixel_values"):
+                delattr(m, "pixel_values")
 
-                if hasattr(m, "pixel_values"):
-                    delattr(m, "pixel_values")
+            if hasattr(m, "marker_cluster"):
```

Comment:
Why isn't it necessary to `delattr pixel_values` anymore?

Reply:
See line 414.